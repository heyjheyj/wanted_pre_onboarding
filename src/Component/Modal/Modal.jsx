import React, { useState } from "react";
import styles from "./Modal.module.css";

const Modal = props => {
  const [isShowModal, setIsShowModal] = useState(false);

  const showModal = () => {
    setIsShowModal(true);
  };

  const hideModal = () => {
    setIsShowModal(false);
  };

  return (
    <div className={styles.container} data-testid="test-container">
      <h3 className={styles.title}>Modal</h3>
      <button
        className={styles.modalButton}
        onClick={showModal}
        data-testid="test-openbutton"
      >
        Open Modal
      </button>
      {isShowModal &&
        <div className={styles.modal} data-testid="test-modal">
          <div className={styles.box}>
            <button onClick={hideModal} data-testid="test-closebutton">
              x
            </button>
            <span>HELLO CODESTATES</span>
          </div>
        </div>}
    </div>
  );
};

export default Modal;
