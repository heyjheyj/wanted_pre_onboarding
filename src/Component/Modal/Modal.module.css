.container {
    width: 80%;
    height: 200px;
    border: 1px solid grey;
    margin-top: 10px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    position: relative;
    align-items: center;
    border-radius: 10px;
}

.title {
    position: absolute;
    top: 5px;
    left: 5px;
}

.modalButton {
    border: none;
    background-color: #431cc6;
    height: 40px;
    color: white;
    font-size: 16px;
    width: 120px;
    border-radius: 25px;
    cursor: pointer;
}

.modal {
    width: 100%;
    height: 100%;
    background-color: #00000080;
    display: flex;
    justify-content: center;
    align-items: center;
    position: absolute;
    border-radius: 8px;
}

.box {
    width: 50%;
    height: 50%;
    background-color: white;
    display: flex;
    flex-direction: column;
    align-items: center;
    border-radius: 10px;
}

.box > button {
    border: none;
    background-color: transparent;
    margin-top: 5px;
    cursor: pointer;
}

.box > span {
    margin-top: 20px;
}