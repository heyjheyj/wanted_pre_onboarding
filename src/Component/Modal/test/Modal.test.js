import Modal from "../Modal";
import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TestRenderer from "react-test-renderer";

describe("Modal", () => {
  it("renders", () => {
    const component = TestRenderer.create(<Modal />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe("Click Event", () => {
    let showModal;
    let openbutton;
    let closebutton;
    let container;
    let modal;

    beforeEach(() => {
      render(<Modal />);
      showModal = jest.fn();
      openbutton = screen.getByTestId("test-openbutton");
      container = screen.getByTestId("test-container");
    });

    it("shows 'modal' when button is clicked", () => {
      userEvent.click(openbutton);

      let modal = screen.getByTestId("test-modal");
      expect(modal).toBeTruthy();
      expect(modal).toBeDefined();
    });

    it("closes 'modal' when 'x'button is clicked", () => {
      userEvent.click(openbutton);
      closebutton = screen.getByTestId("test-closebutton");
      userEvent.click(closebutton);

      const testModal = container.querySelector("div[data-testid=test-modal]");
      expect(testModal).toBeNull();
    });
  });
});
