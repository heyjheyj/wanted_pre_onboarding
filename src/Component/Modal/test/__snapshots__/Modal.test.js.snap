// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders 1`] = `
<div
  className="container"
  data-testid="test-container"
>
  <h3
    className="title"
  >
    Modal
  </h3>
  <button
    className="modalButton"
    data-testid="test-openbutton"
    onClick={[Function]}
  >
    Open Modal
  </button>
</div>
`;
